popx/migration_box.go

```diff
@@ -53,6 +55,57 @@ func WithGoMigrations(migrations Migrations) func(*MigrationBox) *MigrationBox {
 	}
 }
 
+// WithTestdata
+func WithTestdata(testdata fs.FS) func(*MigrationBox) *MigrationBox {
+	testdataPattern := regexp.MustCompile(`^(\d+)_testdata.sql`)
+	return func(m *MigrationBox) *MigrationBox {
```

Comment:
Inside of the option function you don't have a `t *testing.T` parameter, but you can add it to `WithTestdata`. The `fs.WalkDir` collecting all the testdata does not have to happen in the returned function, but you could instead collect the migrations already outside of that and then just append them in the returned function. This will allow you to do `require.NoError(t, err)`.

Reply:
Ah yes, we talked about that :)